File: Java/0203.java
/*Given the head of a linked list and an integer val, remove all the nodes of the linked list that has Node.val == val, and return the new head. */

class ListNode {

  int val;
  ListNode next;

  ListNode() {}

  ListNode(int val) {
    this.val = val;
  }

  ListNode(int val, ListNode next) {
    this.val = val;
    this.next = next;
  }
}

class Solution {

  public ListNode removeElements(ListNode head, int val) {
    ListNode h = new ListNode();
    h.next = head;
    ListNode p = h;
    while (p.next != null) {
      if (p.next.val == val) {
        p.next = p.next.next;
      } else {
        p = p.next;
      }
    }
    return h.next;
  }
}

File: 灵茶/链表/0203.py
from typing import Optional

from 灵茶.链表.ListNode import ListNode

class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode(next=head)
        cur = dummy
        while cur.next:
            if cur.next.val == val:
                cur.next = cur.next.next
            else:
                cur=cur.next
        return dummy.next
